blueocean-dashboard/src/main/java/io/jenkins/blueocean/preload/PipelineRunsStatePreloader.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.blueocean.preload;
+
+import hudson.Extension;
+import hudson.model.Item;
+import io.jenkins.blueocean.commons.BlueUrlTokenizer;
+import io.jenkins.blueocean.commons.RESTFetchPreloader;
+import io.jenkins.blueocean.commons.stapler.ModelObjectSerializer;
+import io.jenkins.blueocean.rest.model.BluePipeline;
+import io.jenkins.blueocean.rest.model.BlueRun;
+import io.jenkins.blueocean.rest.model.BlueRunContainer;
+import io.jenkins.blueocean.service.embedded.rest.BluePipelineFactory;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Preload pipeline runs onto the page if the requested page is a pipeline runs page.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+@Extension
+public class PipelineRunsStatePreloader extends RESTFetchPreloader {
+
+    private static final Logger LOGGER = Logger.getLogger(PipelineRunsStatePreloader.class.getName());
+
+    private static final int DEFAULT_LIMIT = 26;
+
+    @Override
+    protected FetchData getFetchData(@Nonnull BlueUrlTokenizer blueUrl) {
+        BluePipeline pipeline = getPipeline(blueUrl);
+
+        if (pipeline != null) {
+            // It's a pipeline page. Let's prefetch the pipeline runs and add them to the page,
+            // saving the frontend the overhead of requesting them.
+
+            BlueRunContainer runsContainer = pipeline.getRuns();
```

Comment:
@kzantow Actually this wasn't a change in this PR ... just showed up because I renamed the class. Just as well though, otherwise you wouldn't have spotted it. Thanks !!!